src/gtk/toga_gtk/widgets/button.py

```diff
@@ -19,8 +25,23 @@ def set_label(self, label):
     def set_enabled(self, value):
         self.native.set_sensitive(value)
 
-    def set_background_color(self, value):
-        self.interface.factory.not_implemented('Button.set_background_color()')
+    def set_color(self, color):
+        if color:
+            style_context = self.native.get_style_context()
+            css = get_color_css(color)
+            apply(style_context, css)
+
+    def set_background_color(self, color):
+        if color:
+            style_context = self.native.get_style_context()
+            css = get_bg_color_css(color)
+            apply(style_context, css)
+
+    def set_font(self, value):
+        if value:
+            style_context = self.native.get_style_context()
+            css = get_font_css(value)
+            apply(style_context, css)
```

Comment:
Is there any risk to adding these methods to the base widget, rather than explicitly redefining on each widget where we want to support color/font?

Reply:
There is no risk in adding these methods to base widget from my testing. But it does not make sense for widgets like canvas and divider to have font or color. And even in winform back end the implementation is added in each widget class and not in base widget class.
But if you fell like it should be added at base class I have no problems with it.

Reply:
The difference with Winforms is that the Winforms backend requires an explicit "set_background_color()"-style API call, which literally does not exist for canvas/divider etc. As a result, having a "default" implementation will raise a hard error if it is used on a widget that doesn't have the necessary API. 

In the case of GTK, we're setting a style directive; if the widget doesn't support the style, it will be ignored (and, hypothetically, if GTK Canvas ever *did* start to support setting a background color, then it would immediately pick up the style Toga provides). 

That said - thinking about it more, `set_color` et al may need to exist to satisfy the API contract of the backend, in which case moving the implementation to the base class may be a non-starter. My key concern is that, as stands, there's a non-trivial, and essentially identical implementation duplicated across three separate subclasses (and potentially more in future). #1195 describes a general cleanup we'd like to do in this space across all backends; if we're adding new functionality, it makes sense to do this cleanup at the time the feature is added.

Reply:
I will shift the `set_color`, `set_font` & `set_background_color` to base widget class in the next commit.
I would like to point out that you can not set background color to` Gtk.Fixed` (`toga.Box`) widget by this method. According to the research I did about it on the internet, looks like `Gtk.Fixed` does not support changing background color.